pkg/services/authn/authn.go

```diff
@@ -74,6 +74,12 @@ type PostAuthHookFn func(ctx context.Context, identity *Identity, r *Request) er
 type PostLoginHookFn func(ctx context.Context, identity *Identity, r *Request, err error)
 type PreLogoutHookFn func(ctx context.Context, requester identity.Requester, sessionToken *usertoken.UserToken) error
 
+// Could also do away with authn.Client, just not sure why contexthandler needs the whole interface of authn service
```

Comment:
Contexthandler don't need the whole interface just Authenticate from the service. Because it needs to try to authenticate different methods based on what is configured.

But it is a good idea to limit what we need from it. Maybe we should call this `Authenticator` and we can also embed this in `Service`

```go
type Service interface {
  	Authenticator
  	// ... other methods
	
	
}
```